com.unity.shadergraph/Editor/Drawing/Inspector/SGPropertyDrawer.cs

```diff
@@ -0,0 +1,1218 @@
+            propertyFloatField = floatField;
+
+            var defaultRow = new PropertyRow(PropertyDrawerUtils.CreateLabel(labelName, indentLevel));
+            defaultRow.Add(propertyFloatField);
+            defaultRow.styleSheets.Add(Resources.Load<StyleSheet>("Styles/PropertyRow"));
+            return defaultRow;
+        }
+
+        public Action inspectorUpdateDelegate { get; set; }
+
+        public VisualElement DrawProperty(PropertyInfo propertyInfo, object actualObject, Inspectable attribute)
+        {
+            return this.CreateGUI(
+                // Use the setter from the provided property as the callback
+                newValue => propertyInfo.GetSetMethod(true).Invoke(actualObject, new object[] {newValue}),
+                (float) propertyInfo.GetValue(actualObject),
+                attribute.labelName,
+                out var propertyVisualElement);
+        }
+    }
+
+    [SGPropertyDrawer(typeof(Vector2))]
+    class Vector2PropertyDrawer : IPropertyDrawer
+    {
+        internal delegate void ValueChangedCallback(Vector2 newValue);
+
+        public Action preValueChangeCallback { get; set; }
+        public Action postValueChangeCallback { get; set; }
+
+        EventCallback<KeyDownEvent> m_KeyDownCallback;
+        EventCallback<FocusOutEvent> m_FocusOutCallback;
+        public int mUndoGroup { get; set; } = -1;
+
+        public Vector2PropertyDrawer()
+        {
+            CreateCallbacks();
+        }
+
+        void CreateCallbacks()
+        {
+            m_KeyDownCallback = new EventCallback<KeyDownEvent>(evt =>
+            {
+                // Record Undo for input field edit
+                if (mUndoGroup == -1)
+                {
+                    mUndoGroup = Undo.GetCurrentGroup();
+                    preValueChangeCallback?.Invoke();
+                }
+                // Handle escaping input field edit
+                if (evt.keyCode == KeyCode.Escape && mUndoGroup > -1)
+                {
+                    Undo.RevertAllDownToGroup(mUndoGroup);
+                    mUndoGroup = -1;
+                    evt.StopPropagation();
+                }
+                // Dont record Undo again until input field is unfocused
+                mUndoGroup++;
+                postValueChangeCallback?.Invoke();
+            });
+
+            m_FocusOutCallback = new EventCallback<FocusOutEvent>(evt =>
+            {
+                // Reset UndoGroup when done editing input field
+                mUndoGroup = -1;
+            });
+
+        }
+
+        internal VisualElement CreateGUI(
+            ValueChangedCallback valueChangedCallback,
+            Vector2 fieldToDraw,
+            string labelName,
+            out VisualElement propertyVec2Field,
+            int indentLevel = 0)
+        {
+            var vector2Field = new Vector2Field {value = fieldToDraw};
+
+            vector2Field.Q("unity-x-input").Q("unity-text-input").RegisterCallback<KeyDownEvent>(m_KeyDownCallback);
```

Comment:
Queries aren't free, probably want to store the results in a variable to access again (see also other instances of this below).